plugins/backend/catalog-backend-module-okta/src/providers/OktaGroupEntityProvider.test.ts

```diff
@@ -172,5 +177,134 @@ describe('OktaGroupProvider', () => {
+      const provider = OktaGroupEntityProvider.fromConfig(config, {
+        logger,
+        namingStrategy: new ProfileFieldGroupNamingStrategy('org_id')
+          .nameForGroup,
+        userNamingStrategy: 'strip-domain-email',
+      });
+      await provider.connect(entityProviderConnection);
+      await provider.run();
+      expect(entityProviderConnection.applyMutation).toBeCalledWith({
+        type: 'full',
+        entities: expect.arrayContaining([
+          expect.objectContaining({
+            entity: expect.objectContaining({
+              kind: 'Group',
+              metadata: expect.objectContaining({
+                name: '1234',
+              }),
+              spec: expect.objectContaining({
+                members: ['fname'],
+              }),
+            }),
+          }),
+        ]),
+      });
+    });
+
+    it('allows creating a hierarchy for groups', async () => {
+      const entityProviderConnection: EntityProviderConnection = {
+        applyMutation: jest.fn(),
+        refresh: jest.fn(),
+      };
+      const provider = OktaGroupEntityProvider.fromConfig(config, {
+        logger,
+        parentGroupField: 'parent_org_id',
+        namingStrategy: new ProfileFieldGroupNamingStrategy('org_id')
+          .nameForGroup,
+        userNamingStrategy: 'strip-domain-email',
+      });
+      await provider.connect(entityProviderConnection);
+      await provider.run();
+      expect(entityProviderConnection.applyMutation).toBeCalledWith({
+        type: 'full',
+        entities: expect.arrayContaining([
+          expect.objectContaining({
+            entity: expect.objectContaining({
+              kind: 'Group',
+              metadata: expect.objectContaining({
+                name: '1234',
+              }),
+              spec: expect.objectContaining({
+                members: ['fname'],
+                parent: '1234',
+              }),
+            }),
+          }),
+          expect.objectContaining({
+            entity: expect.objectContaining({
+              kind: 'Group',
+              metadata: expect.objectContaining({
+                name: '1235',
+              }),
+              spec: expect.objectContaining({
+                members: ['fname'],
+                parent: '1234',
+              }),
+            }),
+          }),
+        ]),
+      });
+    });
+
+    it('where a failing naming strategy is provided it passes over the group', async () => {
+      const entityProviderConnection: EntityProviderConnection = {
+        applyMutation: jest.fn(),
+        refresh: jest.fn(),
+      };
+      const provider = OktaGroupEntityProvider.fromConfig(config, {
+        logger,
+        namingStrategy: () => {
+          throw new Error('bork');
+        },
+        userNamingStrategy: 'strip-domain-email',
+      });
+      await provider.connect(entityProviderConnection);
+      await provider.run();
+      expect(entityProviderConnection.applyMutation).toBeCalledWith({
```

Comment:
Should this assert `neverHaveBeenCalled` instead of empty array?

Reply:
I think this is correct. Because the provider has found 0 groups so it is going to emit an empty set. This makes sense when you change a filter where previously it was finding more entities than before.